### ISSUE TITLE
Disallow top-level definitions from executing Jobs

### ISSUE THREAD

**Issue by richardxia** [sifive/wake#509]
Because Wake blurs the difference between functions and values, it is easy for newcomers to not understand that a definition without any arguments will execute at the top level. This has a negative, global effect because regardless of which command you are trying to run, these top-level definitions will always execute, even if they run some very long command.

I think we should forbid top-level definitions from executing Jobs at all. If there is some use case for it, then I'd at least like it to be some special thing that must be opted into.

Alternatively, if we just had different keywords for function and variable declarations and disallowed the declaration of a function with zero arguments, I think this would also be fine, as it would match most people's intuitions for the execution model of a programming language.

**Comment by terpstra** [comment]
I do still think some top-level definitions are ok, even ones that call functions:
```
def myOK = 1
def myBad = 2
def myChoices = (myOk, myBad, Nil)
def maxChoice = max myChoices
```
The above all seem like things we want to allow.

I agree that creating Jobs mostly dubious. However, even that can be useful in a pinch.

**Comment by richardxia** [comment]
Yes. Just to be clear, I'm only against heavyweight operations in top-level definitions, such as Jobs. I certainly think that top-level definitions are fine, since in the common case, they're nothing more than global constants.
